core-addon/DataCollection.js

```diff
@@ -0,0 +1,128 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+module.exports = class DataCollection {
+  /**
+   * Sends an empty Ion ping with the provided info.
+   *
+   * @param {String} payloadType
+   *        The type of the encrypted payload. This will define the
+   *        `schemaName` of the ping.
+   *
+   * @param {String} namespace
+   *        The namespace to route the ping. This will define the
+   *        `schemaNamespace` and `studyName` properties of the ping.
+   */
+  async _sendEmptyPing(payloadType, namespace) {
+    // NOTE - while we're not actually sending useful data in
+    // this payload, the current Ion v2 Telemetry pipeline requires
+    // that pings are shaped this way so they are routed to the correct
+    // environment.
+    //
+    // At the moment, the public key used here isn't important but we do
+    // need to use *something*.
+    const publicKey = {
```

Comment:
I wonder if this key should be configurable? It's also still kinda odd to me that we need to do this :)

Reply:
I'd rather hide this, it's really an implementation detail/something we do to make Pipeline happy. But we can revisit this if you feel strongly about it!

Yes, I agree that this is _really odd_. But probably not worth fixing before moving to GleanJS :)